tools/cellxgene_census_builder/src/cellxgene_census_builder/build_state.py

```diff
@@ -20,38 +20,70 @@
 
 CENSUS_BUILD_CONFIG = "config.yaml"
 CENSUS_BUILD_STATE = "state.yaml"
-CENSUS_CONFIG_DEFAULTS = {
-    # General config
-    "verbose": 1,
-    "log_dir": "logs",
-    "log_file": "build.log",
-    "reports_dir": "reports",
-    "consolidate": True,
-    "disable_dirty_git_check": True,
-    "dryrun": False,  # if True, will disable copy of data/logs/reports/release.json to S3 buckets. Will NOT disable local build, etc.
+
+
+@define
+class CensusBuildConfig:
+    verbose: int = field(converter=int, default=1)
+    log_dir: str = field(default="logs")
+    log_file: str = field(default="build.log")
+    reports_dir: str = field(default="reports")
+    consolidate = field(converter=bool, default=True)
+    disable_dirty_git_check = field(converter=bool, default=True)
+    dryrun: bool = field(
+        converter=bool, default=False
+    )  # if True, will disable copy of data/logs/reports/release.json to S3 buckets. Will NOT disable local build, etc.
     #
     # Paths and census version name determined by spec.
-    "cellxgene_census_S3_path": "s3://cellxgene-data-public/cell-census",
-    "logs_S3_path": "s3://cellxgene-data-public-logs/builder",
-    "build_tag": datetime.now().astimezone().date().isoformat(),
+    cellxgene_census_S3_path: str = field(default="s3://cellxgene-data-public/cell-census")
+    cellxgene_census_S3_remote_path: str = field(default=None)
+    logs_S3_path: str = field(default="s3://cellxgene-data-public-logs/builder")
+    build_tag: str = field(default=datetime.now().astimezone().date().isoformat())
     #
     # Default multi-process. Memory scaling based on empirical tests.
-    "multi_process": True,
-    "max_workers": 2 + int(psutil.virtual_memory().total / (96 * 1024**3)),
+    multi_process: bool = field(converter=bool, default=True)
+    max_workers: int = field(default=2 + int(psutil.virtual_memory().total / (96 * 1024**3)))
     #
     # Host minimum resource validation
-    "host_validation_disable": False,  # if True, host validation checks will be skipped
-    "host_validation_min_physical_memory": 512 * 1024**3,  # 512GiB
-    "host_validation_min_swap_memory": 2 * 1024**4,  # 2TiB
-    "host_validation_min_free_disk_space": 1 * 1024**4,  # 1 TiB
+    host_validation_disable: int = field(
+        converter=bool, default=False
+    )  # if True, host validation checks will be skipped
+    host_validation_min_physical_memory: int = field(converter=int, default=512 * 1024**3)  # 512Gi
+    host_validation_min_swap_memory: int = field(converter=int, default=2 * 1024**4)  # 2Ti
```

Comment:
typo in comment - missing B.  Or maybe this is intentional given that they all seem to have changed?   Needs units, so B seems important :-)

Reply:
Nah, I messed up with the editor's multi-cursor. Will fix them accordingly.